lighty-modules/lighty-bgp/src/main/java/io/lighty/modules/bgp/deployer/BgpModule.java

```diff
@@ -0,0 +1,151 @@
+        ribExtensionConsumerContext = createRibExtensions(serializer);
+        peerRegistry = new StrictBGPPeerRegistry();
+        bgpDispatcher = new BGPDispatcherImpl(createBgpExtensions(), bossGroup, workerGroup, peerRegistry);
+        simpleStatementRegistry = createStatementRegistry(dataBroker);
+        routingPolicyFactory = new DefaultBGPRibRoutingPolicyFactory(
+                dataBroker, simpleStatementRegistry);
+        codecsRegistry = new ConstantCodecsRegistry(codecTree);
+        tableTypeRegistryConsumer = createBgpTableTypes();
+        stateCollector = new BGPStateCollector();
+        stateProvider = new StateProviderImpl(dataBroker, 5, tableTypeRegistryConsumer, stateCollector,
+                DEFAULT_BGP_NETWORK_INSTANCE_NAME);
+        bgpDeployer = new DefaultBgpDeployer(DEFAULT_BGP_NETWORK_INSTANCE_NAME, cssProvider, rpcProvider,
+                ribExtensionConsumerContext, bgpDispatcher, routingPolicyFactory, codecsRegistry, domDataBroker,
+                dataBroker, tableTypeRegistryConsumer, stateCollector);
+    }
+
+    public BgpModule(final EffectiveModelContext modelContext, final DataBroker dataBroker,
+            final DOMDataBroker domDataBroker, final BindingCodecTreeFactory codecTreeFactory,
+            final BindingRuntimeContext runtimeContext, final RpcProviderService rpcProvider,
+            final ClusterSingletonServiceProvider cssProvider, final BindingNormalizedNodeSerializer serializer,
+            final EventLoopGroup bossGroup, final EventLoopGroup workerGroup) {
+        this(modelContext, dataBroker, domDataBroker, codecTreeFactory.create(runtimeContext),
+                rpcProvider, cssProvider, serializer, bossGroup, workerGroup);
+    }
+
+    public BgpModule(final LightyServices lightyServices) {
+        this(lightyServices.getEffectiveModelContextProvider().getEffectiveModelContext(),
+                lightyServices.getBindingDataBroker(), lightyServices.getClusteredDOMDataBroker(),
+                lightyServices.getBindingCodecTreeFactory(),
+                lightyServices.getAdapterContext().currentSerializer().getRuntimeContext(),
+                lightyServices.getRpcProviderService(), lightyServices.getClusterSingletonServiceProvider(),
+                lightyServices.getBindingNormalizedNodeSerializer(), lightyServices.getBossGroup(),
+                lightyServices.getWorkerGroup());
+    }
+
+
+    @Override
+    @SuppressWarnings({"checkstyle:illegalCatch"})
+    protected boolean initProcedure() {
+        try {
+            //Load initial routing policies into datastore
+            initialConfigLoader.init();
+            simpleStatementRegistry.start();
+            bgpDeployer.init();
+            return true;
+        } catch (Exception e) {
+            LOG.warn("Failed to initialize BGPModule", e);
+            return false;
+        }
+    }
+
+    @Override
+    @SuppressWarnings({"checkstyle:illegalCatch"})
+    protected boolean stopProcedure() {
```

Comment:
Please close instances in separate try-catch block to make sure that every instance will be closed if one will fail.

Reply:
done